src/service/resource.js

```diff
@@ -1030,10 +1048,22 @@ export class Resource {
   unlayout() {
     if (
       this.state_ == ResourceState.NOT_BUILT ||
-      this.state_ == ResourceState.NOT_LAID_OUT
+      this.state_ == ResourceState.NOT_LAID_OUT ||
+      this.state_ == ResourceState.READY_FOR_LAYOUT
     ) {
       return;
     }
+    if (this.abortController_) {
+      this.abortController_.abort();
```

Comment:
Do we also need to set `this.abortController_ = null` here or if layout completes successfuly?

Reply:
Done.